Sources/Xcodeproj/Module+PBXProj.swift

```diff
@@ -298,3 +291,21 @@ extension Module  {
         return buildSettings
     }
 }
+
+extension Plist {
+    // FIXME: This is only internal for unit testing.
+    /// Quotes the strings in the Plist.
+    func quoteStrings() -> Plist {
+        switch self {
+        case .string(let str):
+            return .string("\"" + str + "\"")
```

Comment:
So rather than have the business logic deal with plist node types, I would keep build settings as a mapping from build setting names to values (each of which is either a string or a string array).

Then when serializing you go through it; any value that's already a string is unchanged, and any value that's an array of strings is turned into a string by quoting each array element and joining with whitespace.  And then wrapped in a `.string()` plist node.